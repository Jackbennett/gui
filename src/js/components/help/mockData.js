export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
