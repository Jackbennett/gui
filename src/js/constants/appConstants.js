module.exports = {
  SET_SNACKBAR: 'SET_SNACKBAR',
  RECEIVED_HOSTED_LINKS: 'RECEIVED_HOSTED_LINKS',
  SET_ANNOUNCEMENT: 'SET_ANNOUNCEMENT',
  SET_FIRST_LOGIN_AFTER_SIGNUP: 'SET_FIRST_LOGIN_AFTER_SIGNUP',
  PLANS: {
    os: {
      name: 'Basic',
      offer: true,
      price: '$29/month',
      deviceCount: 'for first 50 devices',
      offerprice: '$23/month for first 50 devices',
      price2: 'for first 6 months;\n$29/month thereafter',
      features: ['Basic OTA features'],
      value: 'os'
    },
    professional: {
      name: 'Professional',
      offer: true,
      price: '$249/month',
      deviceCount: 'for first 250 devices',
      offerprice: '$200/month for first 50 devices',
      price2: 'for first 6 months;\n$249/month thereafter',
      features: ['+ Advanced OTA features', '+ Standard support'],
      value: 'professional'
    },
    enterprise: {
      name: 'Enterprise',
      price: 'Custom pricing',
      deviceCount: 'unlimited devices',
      features: ['+ Enterprise-grade OTA features', '+ Premium support'],
      value: 'enterprise'
    }
  },
  ADDONS: {
    configure: {
      title: 'Configure',
      description: 'Expand your plan with device configuration features',
      os: {
        price: '$10/month',
        deviceCount: 'for first 50 devices'
      },
      professional: {
        price: '$60/month',
        deviceCount: 'for first 250 devices'
      }
    },
    troubleshoot: {
      title: 'Troubleshoot',
      description: 'Expand your plan with device troubleshooting features',
      os: {
        price: '$24/month',
        deviceCount: 'for first 50 devices'
      },
      professional: {
        price: '$67/month',
        deviceCount: 'for first 250 devices'
      }
    }
  }
};
