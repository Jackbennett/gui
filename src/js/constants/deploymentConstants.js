module.exports = {
  CREATE_DEPLOYMENT: 'CREATE_DEPLOYMENT',
  REMOVE_DEPLOYMENT: 'REMOVE_DEPLOYMENT',
  RECEIVE_DEPLOYMENT: 'RECEIVE_DEPLOYMENT',
  RECEIVE_DEPLOYMENT_STATS: 'RECEIVE_DEPLOYMENT_STATS',
  RECEIVE_DEPLOYMENT_DEVICE_LOG: 'RECEIVE_DEPLOYMENT_DEVICE_LOG',
  RECEIVE_DEPLOYMENT_DEVICES: 'RECEIVE_DEPLOYMENT_DEVICES',
  RECEIVE_DEPLOYMENTS: 'RECEIVE_DEPLOYMENTS',
  RECEIVE_PENDING_DEPLOYMENTS: 'RECEIVE_PENDING_DEPLOYMENTS',
  RECEIVE_INPROGRESS_DEPLOYMENTS: 'RECEIVE_INPROGRESS_DEPLOYMENTS',
  RECEIVE_SCHEDULED_DEPLOYMENTS: 'RECEIVE_SCHEDULED_DEPLOYMENTS',
  RECEIVE_FINISHED_DEPLOYMENTS: 'RECEIVE_FINISHED_DEPLOYMENTS',
  RECEIVE_PENDING_DEPLOYMENTS_COUNT: 'RECEIVE_PENDING_DEPLOYMENTS_COUNT',
  RECEIVE_INPROGRESS_DEPLOYMENTS_COUNT: 'RECEIVE_INPROGRESS_DEPLOYMENTS_COUNT',
  RECEIVE_SCHEDULED_DEPLOYMENTS_COUNT: 'RECEIVE_SCHEDULED_DEPLOYMENTS_COUNT',
  RECEIVE_FINISHED_DEPLOYMENTS_COUNT: 'RECEIVE_FINISHED_DEPLOYMENTS_COUNT',
  SELECT_INPROGRESS_DEPLOYMENTS: 'SELECT_INPROGRESS_DEPLOYMENTS',
  SELECT_PENDING_DEPLOYMENTS: 'SELECT_PENDING_DEPLOYMENTS',
  SELECT_SCHEDULED_DEPLOYMENTS: 'SELECT_SCHEDULED_DEPLOYMENTS',
  SELECT_FINISHED_DEPLOYMENTS: 'SELECT_FINISHED_DEPLOYMENTS',
  SELECT_DEPLOYMENT: 'SELECT_DEPLOYMENT'
};
