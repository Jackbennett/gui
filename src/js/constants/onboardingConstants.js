module.exports = {
  SET_ONBOARDING_COMPLETE: 'SET_ONBOARDING_COMPLETE',
  SET_ONBOARDING_PROGRESS: 'SET_ONBOARDING_PROGRESS',
  SET_ONBOARDING_DEVICE_TYPE: 'SET_ONBOARDING_DEVICE_TYPE',
  SET_ONBOARDING_APPROACH: 'SET_ONBOARDING_APPROACH',
  SET_ONBOARDING_ARTIFACT_INCLUDED: 'SET_ONBOARDING_ARTIFACT_INCLUDED',
  SET_SHOW_CREATE_ARTIFACT: 'SET_SHOW_CREATE_ARTIFACT',
  SET_SHOW_ONBOARDING_HELP: 'SET_SHOW_ONBOARDING_HELP',
  SET_SHOW_ONBOARDING_HELP_DIALOG: 'SET_SHOW_ONBOARDING_HELP_DIALOG'
};
