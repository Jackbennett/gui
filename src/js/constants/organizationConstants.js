module.exports = {
  AUDIT_LOGS_TYPES: [
    { title: 'Deployment', value: 'deployment' },
    { title: 'User', value: 'user' }
  ],
  RECEIVE_AUDIT_LOGS: 'RECEIVE_AUDIT_LOGS',
  SET_ORGANIZATION: 'SET_ORGANIZATION'
};
