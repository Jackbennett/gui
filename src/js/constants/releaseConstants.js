module.exports = {
  ARTIFACT_GENERATION_TYPE: {
    SINGLE_FILE: 'single_file'
  },
  ARTIFACTS_REMOVED_ARTIFACT: 'ARTIFACTS_REMOVED_ARTIFACT',
  ARTIFACTS_SET_ARTIFACT_URL: 'ARTIFACTS_SET_ARTIFACT_URL',
  UPLOAD_ARTIFACT: 'UPLOAD_ARTIFACT',
  UPLOAD_PROGRESS: 'UPLOAD_PROGRESS',
  UPDATED_ARTIFACT: 'UPDATED_ARTIFACT',
  RECEIVE_ARTIFACTS: 'RECEIVE_ARTIFACTS',
  RECEIVE_RELEASE: 'RECEIVE_RELEASE',
  RECEIVE_RELEASES: 'RECEIVE_RELEASES',
  RELEASE_REMOVED: 'RELEASE_REMOVED',
  SHOW_REMOVE_DIALOG: 'SHOW_REMOVE_DIALOG',
  SELECTED_ARTIFACT: 'SELECTED_ARTIFACT',
  SELECTED_RELEASE: 'SELECTED_RELEASE'
};
