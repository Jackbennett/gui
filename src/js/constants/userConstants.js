const apiUrl = '/api/management/v1';
const useradmApiUrl = `${apiUrl}/useradm`;

module.exports = {
  RECEIVED_QR_CODE: 'RECEIVED_QR_CODE',

  SUCCESSFULLY_LOGGED_IN: 'SUCCESSFULLY_LOGGED_IN',
  USER_LOGOUT: 'USER_LOGOUT',
  RECEIVED_USER_LIST: 'RECEIVED_USER_LIST',
  RECEIVED_USER: 'RECEIVED_USER',
  CREATED_USER: 'CREATED_USER',
  REMOVED_USER: 'REMOVED_USER',
  UPDATED_USER: 'UPDATED_USER',

  RECEIVED_ROLES: 'RECEIVED_ROLES',
  CREATED_ROLE: 'CREATED_ROLE',
  UPDATED_ROLE: 'UPDATED_ROLE',
  REMOVED_ROLE: 'REMOVED_ROLE',

  SET_CURRENT_USER: 'SET_CURRENT_USER',
  SET_GLOBAL_SETTINGS: 'SET_GLOBAL_SETTINGS',
  SET_SHOW_HELP: 'SET_SHOW_HELP',
  SET_SHOW_CONNECT_DEVICE: 'SET_SHOW_CONNECT_DEVICE',

  emptyRole: { title: undefined, allowUserManagement: false, groups: [], description: '', editable: undefined, permissions: [] },
  rolesByName: {
    admin: 'RBAC_ROLE_PERMIT_ALL',
    readOnly: 'RBAC_ROLE_OBSERVER',
    ci: 'RBAC_ROLE_CI',
    userManagement: { action: 'http', object: { type: 'any', value: `${useradmApiUrl}/.*` } },
    deploymentCreation: { action: 'CREATE_DEPLOYMENT', object: { type: 'DEVICE_GROUP', value: undefined } },
    groupAccess: { action: 'VIEW_DEVICE', object: { type: 'DEVICE_GROUP', value: undefined } }
  },
  useradmApiUrl
};
