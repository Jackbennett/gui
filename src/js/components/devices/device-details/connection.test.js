import React from 'react';
import { render } from '@testing-library/react';
import DeviceConnection from './connection';
import { defaultState, undefineds } from '../../../../../tests/mockData';
import { DEVICE_CONNECT_STATES } from '../../../constants/deviceConstants';

describe('DeviceConnection Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(<DeviceConnection device={defaultState.devices.byId.a1} setSnackbar={jest.fn} />);
    const view = baseElement.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
  it('renders correctly when disconnected', async () => {
    const { baseElement } = render(
      <DeviceConnection device={{ ...defaultState.devices.byId.a1, connect_status: DEVICE_CONNECT_STATES.disconnected }} setSnackbar={jest.fn} />
    );
    const view = baseElement.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
  it('renders correctly when connected', async () => {
    const { baseElement } = render(
      <DeviceConnection device={{ ...defaultState.devices.byId.a1, connect_status: DEVICE_CONNECT_STATES.connected }} setSnackbar={jest.fn} />
    );
    const view = baseElement.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
