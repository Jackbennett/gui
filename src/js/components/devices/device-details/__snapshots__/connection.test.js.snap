// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceConnection Component renders correctly 1`] = `
<div>
  <div
    class="device-connect bordered report-list"
  >
    <h4
      class="margin-bottom-small"
    >
      Remote Terminal
    </h4>
    <div
      class="flexbox"
      style="flex-direction: row;"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
        style="text-transform: none; text-align: left;"
      >
        The Remote terminal add-on does not seem to be enabled on this device.
        <br />
        Please see
         
        <a
          href="https://docs.mender.io/add-ons/remote-terminal"
          rel="noopener noreferrer"
          target="_blank"
        >
          the documentation
        </a>
         
        for a description on how it works and how to enable it.
      </p>
    </div>
  </div>
</div>
`;
